apps/emqx/src/emqx_persistent_message_ds_replayer.erl

```diff
@@ -176,9 +177,12 @@ fetch(SessionId, Inflight0, [DSStream | Streams], N, Acc) when N > 0 ->
     #inflight{next_seqno = FirstSeqno, offset_ranges = Ranges} = Inflight0,
     ItBegin = get_last_iterator(DSStream, Ranges),
     {ok, ItEnd, Messages} = emqx_ds:next(?PERSISTENT_MESSAGE_DB, ItBegin, N),
-    {Publishes, UntilSeqno} = publish(FirstSeqno, Messages),
-    case range_size(FirstSeqno, UntilSeqno) of
-        Size when Size > 0 ->
+    case Messages of
+        [] ->
+            fetch(SessionId, Inflight0, Streams, N, Acc);
```

Comment:
Update iterator in `Inflight0`?

Reply:
I'd say it's an optimization rather than feature. I'll create a follow-up ticket.